src/root.ts

```diff
@@ -1,5 +1,11 @@
+import log2 from './log2';
+
 function rootPos(x: bigint, n: bigint): bigint {
-  var a = x, b = x+1n, m = n-1n;
+  const initialGuess = 1n << ((log2(x) / n) + 1n);
+  if (initialGuess === 2n) {
```

Comment:
Is there an edge case here?

Reply:
my tests fail with "RangeError: Maximum BigInt size exceeded" without this branch